OpenSim/Common/Path.h

```diff
@@ -0,0 +1,162 @@
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "OpenSim/Common/Object.h"
+
+namespace OpenSim {
+
+//==============================================================================
+//                                 OPENSIM PATH
+//==============================================================================
+/**
+* An abstract class for handling a Path. A Path refers to a list of strings that
+* represent a different level in a hierarchical structure. Each level is divided
+* by designated separators (e.g., "/" or "\"). The string name related to each
+* level is called a "pathElement" here.
+*
+* This class stores the list of levels as a vector of strings. One char is used
+* to denote a path separator when either reading in or writing the Path out to
+* a string. A bool is stored to determine whether this Path should be resolved
+* relative to the root (denoted with a leading separator char) or not.
+*
+* @author Carmichael Ong
+*/
+class OSIMCOMMON_API Path {
+
+public:
+    /// Default constructor
+    Path() = default;
+
+    /// Construct Path from a string, given separator character and a string
+    /// of invalidChars. Performs a cleanPath() at the end of construction.
+    Path(const std::string path,
+         const char separator,
+         const std::string invalidChars);
+
+    /// Construct Path from a vector of strings (pathVec), given separator
+    /// character and a string of invalidChars. Performs a cleanPath() at 
+    /// the end of construction.
+    Path(std::vector<std::string> pathVec,
+         const char separator,
+         const std::string invalidChars,
+         bool isAbsolute);
+
+    /// Use default copy constructor and assignment operator.
+    Path(const Path&) = default;
+    Path& operator=(const Path&) = default;
+
+    /// Destructor.
+    ~Path() = default;
+
+    /// Write out the path to a string with each element separated by the
+    /// specified separator.
+    std::string getString() const;
+
+    /// Return true if this Path is an absolute path.
+    bool isAbsolute() { return _isAbsolute; };
+
+    /// Return the number of elements in the Path
+    size_t getPathLength() const { return _path.size(); };
+
+    /// Push a string to the back of a path (i.e. the end). First checks if the
+    /// pathElement is legal. Does not change if path is absolute or relative.
```

Comment:
`legal` -> `valid`

Consider: `Does not change if path is absolute or relative.` -> `This function does not alter whether this path is absolute or relative.`